enterprise_access/apps/content_assignments/tests/test_utils.py

```diff
@@ -192,3 +236,41 @@ def test_get_self_paced_normalized_start_date_self_paced(self, start_date, end_d
         else:
             assert get_self_paced_normalized_start_date(start_date, end_date, course_metadata) == \
                    start_date
+
+    @ddt.data(
+        # Expects the content_key for an is_assigned_course_run
+        {
+            'assignment': {
+                'is_assigned_course_run': True,
+                'preferred_course_run_key': 'course-v1:edX+DemoX+1T360',
+                'content_key': 'course-v1:edX+DemoX+1T60',
+            },
+            'expected_output': mock_course_run_2,
+        },
+        # Expects the preferred content_key
+        {
+            'assignment': {
+                'is_assigned_course_run': False,
+                'preferred_course_run_key': 'course-v1:edX+DemoX+1T360',
+                'content_key': 'edX+DemoX',
+            },
+            'expected_output': mock_course_run_1,
+        },
+        # Expects the top level course key
+        {
+            'assignment': {
+                'is_assigned_course_run': False,
+                'preferred_course_run_key': None,
+                'content_key': 'edX+DemoX',
+            },
+            'expected_output': mock_advertised_course_run,
+        },
+    )
+    @ddt.unpack
+    def test_get_normalized_metadata_for_assignment(self, assignment, expected_output):
+        assignment_obj = SimpleNamespace(**assignment)
```

Comment:
Might recommend using `LearnerContentAssignmentFactory` versus creating a pseudo-assignment instance based on the `assignment` dict in the test case data using `SimpleNamespace`. This would more accurately reflect how the util function is used in practice.

```python
from enterprise_access.apps.content_assignments.tests.factories import LearnerContentAssignmentFactory
assignment_obj = LearnerContentAssignmentFactory(**assignment)
```